### PR TITLE
8261673: Move javadoc for the lookup mechanism to module-info

### PR DESCRIPTION
**Pull request by JoeWang-Java**
Consolidate and move javadoc for the lookup mechanism to the module summary.

/csr

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261673](https://bugs.openjdk.java.net/browse/JDK-8261673): Move javadoc for the lookup mechanism to module-info

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)

### Download
To checkout this PR locally:
`$ git fetch https://git.openjdk.java.net/jdk pull/3020/head:pull/3020`
`$ git checkout pull/3020`

To update a local copy of the PR:
`$ git checkout pull/3020`
`$ git pull https://git.openjdk.java.net/jdk pull/3020/head`

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]

File: src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java [hunk 1]
```diff
@@ -144,55 +144,11 @@ public static XMLOutputFactory newInstance()
...
-   * </li>
-   * <li>
-   *   <p>
-   *   Otherwise, the {@linkplain #newDefaultFactory() system-default}
-   *   implementation is returned.
-   * </li>
-   * </ul>
+   * Creates a new instance of the factory. This method uses the
+   * <a href="../../../module-summary.html#LookupMechanism">JAXP Lookup Mechanism</a>
+   * to determine the {@code XMLEventFactory} implementation class to load.
```
Comment (naotoj) [inline src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java:149]: And this one as `XMLOutputFactory`?

**Review by LanceAndersen — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

File: src/java.xml/share/classes/javax/xml/stream/XMLInputFactory.java [hunk 2]
```diff
@@ -167,53 +167,12 @@ public static XMLInputFactory newInstance()
...
-   *   ClassLoader#getSystemClassLoader() system class loader} will be used.
-   * </li>
-   * <li>
-   * <p>Otherwise, the {@linkplain #newDefaultFactory() system-default}
-   *    implementation is returned.
-   * </li>
-   * </ul>
+   * Creates a new instance of the factory. This method uses the
+   * <a href="../../../module-summary.html#LookupMechanism">JAXP Lookup Mechanism</a>
+   * to determine the {@code XMLEventFactory} implementation class to load.
```
[the remark this answers is not in the document]
Comment (JoeWang-Java) [reply src/java.xml/share/classes/javax/xml/stream/XMLInputFactory.java]: Fixed. Updated CSR as well.

File: src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java (hunk 1 above)
Comment (JoeWang-Java) [reply src/java.xml/share/classes/javax/xml/stream/XMLOutputFactory.java]: Fixed. Thanks.

**Review by irisclark — APPROVED** [review]
Nice consolidation.  I've added myself as a reviewer of the CSR too.

**Pull request closed without merging**